SFS_ir/get_ir.m

```
@@ -124,28 +140,53 @@
-    [xs(1),xs(2),xs(3)] = cart2sph(xs(1),xs(2),xs(3));
+    % Get measured loudspeaker positions
+    x0 = sofa_get_secondary_sources(header,'cartesian');
+    x0 = x0(:,1:3); % need position only
     % Combine head orientation and desired direction of source (see note above)
+    [xs(1),xs(2),xs(3)] = cart2sph(xs(1),xs(2),xs(3));
     xs(1) = correct_azimuth(xs(1)-head_orientation(1));
     xs(2) = correct_elevation(xs(2)-head_orientation(2));
-    % Find nearest neighbours and interpolate if desired and needed
-    [neighbours,idx] = findnearestneighbour(x0(:,1:2)',xs(1:2),3);
+    [xs(1),xs(2),xs(3)] = sph2cart(xs(1),xs(2),xs(3));
+    % Get nearest neighbour point or points for interpolation
+    if useinterpolation
+        switch interpolationpointselection
+        case 'nearestneighbour'
+            [idx,weights] = findnearestneighbour(x0,xs,2);
+        case 'delaunay'
+            [idx,weights] = findconvexcone(x0,xs);
+        case 'voronoi'
+            to_be_implemented;
+        otherwise
+            error(['%s: ''%s'' is an unknown method to select interpolation ', ...
+                'points.'],upper(mfilename),interpolationpointselection);
+        end
+    else
+        [idx,weights] = findnearestneighbour(x0,xs,1);
+    end
+    % Get impulse responses according to point selection
     ir = sofa_get_data_fir(sofa,idx);
+    % Correct distance
+    [x0(:,1),x0(:,2),x0(:,3)] = cart2sph(x0(:,1),x0(:,2),x0(:,3));
+    [xs(1),xs(2),xs(3)] = cart2sph(xs(1),xs(2),xs(3));
     ir = ir_correct_distance(ir,x0(idx,3),xs(3),conf);
-    [ir,x0] = interpolate_ir(ir,neighbours,xs(1:2)',conf);
-    [x0(1),x0(2),x0(3)] = sph2cart(x0(1),x0(2),xs(3));
+    [x0(:,1),x0(:,2),x0(:,3)] = sph2cart(x0(:,1),x0(:,2),x0(:,3));
+    % Select or interpolate to desired impulse response
+    [ir,weights_selected,x0_selected] = interpolate_ir(ir,weights,x0(idx,:),conf);
+    % Calculate position of returned impulse response
+    x0_new = weights_selected'*x0_selected;
```

OK.
What will happen if the two IRs used for interpolation don't have the same radius?
Then the `simple` interpolation method makes no sense, but will still return something. What is with the `freqdomain` method, is this working with different radii?
Otherwise we should maybe return an error if the radii are not the same. And I guess then we could also move the corrected calculation of `x0_new` to `interpolate_ir()`.